[receiver/github] add repository custom properties as resource attributes (#40878)
open-telemetry/opentelemetry-collector-contrib: receiver/githubreceiver/README.md

[run]: https://github.com/krzko/run-with-telemetry
[otcli]: https://github.com/equinix-labs/otel-cli
[tr]: ./trace_event_handling.go

### Custom Properties as Resource Attributes

> The GitHub receiver supports adding custom properties from GitHub repositories as resource attributes in your telemetry data. This feature allows you to enrich your traces with additional metadata specific to your repositories.

Reviewer: ```suggestion
The GitHub receiver supports adding custom properties from GitHub repositories as resource attributes in your telemetry data. This allows users to enrich traces and events with additional metadata specific to each repository.
```

#### How It Works

When a GitHub webhook event is received, the receiver extracts all custom properties from the repository and adds them as resource attributes with the prefix `github.repository.custom_properties`.

For example, if your repository has these custom properties:

```
team-name: platform-engineering

> environment: production

Reviewer: ```suggestion
```
Let's replace the environment example with something else. Mainly because this could cause confusion with the semantic convention for deployment environments which is `deployment.environment.name` which really should come from deployment events and may not be exclusive to a single repository. Really I just want to avoid confusion on this.

service-tier: critical
```

They will be added as resource attributes:

```
github.repository.custom_properties.team_name: "platform-engineering"

> github.repository.custom_properties.environment: "production"

Author: That makes sense. I altered the examples in there.
